src/app/core/shared/operators.ts

```diff
@@ -178,18 +179,29 @@ export const getAllSucceededRemoteListPayload = () =>
  * Operator that checks if a remote data object returned a 401 or 404 error
  * When it does contain such an error, it will redirect the user to the related error page, without altering the current URL
  * @param router The router used to navigate to a new page
+ * @param authService Service to check if the user is authenticated
  */
-export const redirectOn404Or401 = (router: Router) =>
+export const redirectOn4xx = (router: Router, authService: AuthService) =>
   <T>(source: Observable<RemoteData<T>>): Observable<RemoteData<T>> =>
-    source.pipe(
-      tap((rd: RemoteData<T>) => {
+    observableCombineLatest(source, authService.isAuthenticated()).pipe(
+      map(([rd, isAuthenticated]: [RemoteData<T>, boolean]) => {
         if (rd.hasFailed) {
           if (rd.error.statusCode === 404) {
             router.navigateByUrl(getPageNotFoundRoute(), {skipLocationChange: true});
-          } else if (rd.error.statusCode === 401) {
-            router.navigateByUrl(getUnauthorizedRoute(), {skipLocationChange: true});
+          } else if (rd.error.statusCode === 403 || rd.error.statusCode === 401) {
+            if (isAuthenticated) {
+              if (rd.error.statusCode === 403) {
+                router.navigateByUrl(getForbiddenRoute(), {skipLocationChange: true});
+              } else if (rd.error.statusCode === 401) {
+                router.navigateByUrl(getUnauthorizedRoute(), {skipLocationChange: true});
```

Comment:
Silly question perhaps, but how can a 401 ever occur when you are already authenticated (i.e. `if (isAuthenticated)`)?  If that ever occurs, that'd imply a major flaw in the REST API -- as a 401 means you are *not authenticated* or failed to authenticate. 

Do we even need `getUnauthorizedRoute()` anymore, or is this just a safety measure of some sort?  It also makes me wonder if we need the `UnauthorizedComponent` anymore, as it seems like it should never be used.

Reply:
You're right, we'll remove it